### PR TITLE
Added CONTRIBUTING.md

### LINKED ISSUE #2651: Add CONTRIBUTING.md
## 🚀 Feature

It is a good practice to have CONTRIBUTING.md file to describe some guidelines for open-source contributions.

TODO:
- [x] Version 1
- [x] complete section about how to add a new model
- [ ] complete section about how to add a new dataset

cc @fmassa

### PR DESCRIPTION
**Pull request by vfdev-5**
Related #2651

Description:

- Added a draft for CONTRIBUTING.md

This PR is not closing the issue #2651 as there are 2 sections to be defined later.

EDIT: There is already a PR submitted by the bot : https://github.com/pytorch/vision/pull/1564
I was not aware before.

### CODE REVIEW

**Comment by datumbox** [comment]
Any chance we could merge this to master? It's a good source of info for new comers as it describes setting up the dev environment. We could continue improving it post merge.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+```
+Then open the browser at `0.0.0.0:<port>` (e.g. `0.0.0.0:1234`)
+
+### New model
+
+TDB
+ 
+### New dataset
+
+TDB
```
Comment (jgbradley1) [inline CONTRIBUTING.md:135]: Shouldn't this be TBD (to be determined)?

**Comment by jgbradley1** [comment]
This is just a suggestion. Should new datasets support at a minimum, the same constructor arguments (when appropriate)?

* root
* split
* target_type
* target_transform
* transform
* download - for large datasets that are unreasonable to download, should always raise a RuntimeError when `download=True`

**Comment by datumbox** [comment]
@vfdev-5 @fmassa I really think we should merge this, even if there are a few things left to TBD. This doc contains useful information that will allow new contributors to learn how to build the library from source. What is the bare minimum we need to do to merge this?

**Review by fmassa — APPROVED** [review]
Thanks a lot for working on this @vfdev-5 !

Let's get this merged, and at the same time can you create an issue (or keep the existing one open) so that we track completing the points about datasets / models?

**Comment by fmassa** [comment]
@jgbradley1 that's a good suggestion, although I'm not sure it can be generically implemented everywhere. FYI there was some discussion already in the past about dataset standardization in https://github.com/pytorch/vision/issues/1080

**Comment by vfdev-5** [comment]
@fmassa let's keep existing issue open and I add a message about that.

**Review by pmeier — APPROVED** [review]

**Pull request merged**